packages/firestore/test/integration/api/validation.test.ts

```diff
@@ -117,45 +120,61 @@ apiDescribe('Validation:', (persistence: boolean) => {
       return;
     }
 
-    validationIt(persistence, 'validates options', db => {
-      // NOTE: 'credentials' is an undocumented API so ideally we wouldn't
-      // show it in the error, but I don't think it's worth the trouble of
-      // hiding it.
-      expect(() => db.settings({ invalid: true } as any)).to.throw(
-        "Unknown option 'invalid' passed to function settings(). " +
-          'Available options: host, ssl, credentials'
-      );
-
-      expect(() =>
-        db.settings({
-          ssl: true
-        })
-      ).to.throw("Can't provide ssl option if host option is not set");
+    (validatesJsInput ? validationIt : validationIt.skip)(
+      persistence,
+      'validates options',
+      db => {
+        // NOTE: 'credentials' is an undocumented API so ideally we wouldn't
+        // show it in the error, but I don't think it's worth the trouble of
+        // hiding it.
+        expect(() => db.settings({ invalid: true } as any)).to.throw(
+          "Unknown option 'invalid' passed to function settings(). " +
+            'Available options: host, ssl, credentials'
+        );
 
-      expect(() => db.settings({ host: null as any })).to.throw(
-        'Function settings() requires its host option to be of type ' +
-          'non-empty string, but it was: null'
-      );
-    });
+        expect(() =>
+          db.settings({
+            ssl: true
+          })
+        ).to.throw("Can't provide ssl option if host option is not set");
+
+        expect(() => db.settings({ host: null as any })).to.throw(
+          'Function settings() requires its host option to be of type ' +
+            'non-empty string, but it was: null'
+        );
+      }
+    );
 
     validationIt(persistence, 'disallows changing settings after use', db => {
+      let errorMsg =
+        'Firestore has already been started and its settings can no ' +
+        'longer be changed. ';
+
+      if (usesModularApi()) {
+        errorMsg +=
+          'initializeFirestore() cannot be called after calling ' +
+          'getFirestore()';
+      } else {
+        errorMsg +=
+          'You can only call settings() before calling any other ' +
+          'methods on a Firestore object.';
+      }
+
       db.doc('foo/bar');
       expect(() =>
         db.settings({ host: 'something-else.example.com' })
-      ).to.throw(
-        'Firestore has already been started and its settings can no ' +
-          'longer be changed. You can only call settings() before ' +
-          'calling any other methods on a Firestore object.'
-      );
+      ).to.throw(errorMsg);
     });
 
-    validationIt(persistence, 'enforces minimum cache size', db => {
+    validationIt(persistence, 'enforces minimum cache size', () => {
+      const db = newTestFirestore('testProjectId');
```

Comment:
This is an invalid project ID. Only lowercase, numbers, and dashes are allowed.

Reply:
Uses "test-project" here and elsewhere (via a new TEST_PROJECT export).